packages/protocol/contracts/common/Proxy.sol

```diff
@@ -88,6 +88,7 @@ contract Proxy {
     payable
     onlyOwner
   {
+    require(isContract(implementation), "Invalid contract address");
```

Comment:
Wouldn't this need to be checked on every proxied call, in case the implementation contract self destructs?

Reply:
added to fallback function